adobe_vipm/flows/airtable.py

```diff
@@ -429,3 +511,113 @@ def get_prices_for_3yc_skus(product_id, currency, start_date, skus):
+        AirTableBaseInfo.for_migrations(product_id)
+    )
+    GlobalCustomerAgreementDeployment.batch_save(
+        [
+            GlobalCustomerAgreementDeployment(**agreement_deployment)
+            for agreement_deployment in agreement_deployments
+        ]
+    )
+
+
+def create_global_customer_main_agreement(product_id, main_agreement):
+    """
+    Create a GlobalCustomerMainAgreement object.
+
+    Args:
+        product_id (str): The ID of the product used to determine the AirTable base.
+        main_agreement (dict): The main agreement object to create.
+    """
+    GlobalCustomerMainAgreement = get_global_customer_main_agreement_model(
+        AirTableBaseInfo.for_migrations(product_id)
+    )
+    GlobalCustomerMainAgreement(**main_agreement).save()
+
+
+def get_global_customer_main_agreement(
+    product_id, authorization_uk, membership_or_customer_id
+):
+    """
+    Retrieve a GlobalCustomerMainAgreement object given the authorization ID and
+    the membership ID or the customer ID.
+
+    Args:
+        product_id (str): The ID of the product used to determine the AirTable base.
+        authorization_uk (str): The ID of the authorization.
+        membership_or_customer_id (str): Either a membership ID or a customer ID.
+
+    Returns:
+        GlobalCustomerMainAgreement: The GlobalCustomerMainOrder if it has been found,
+        None otherwise.
+    """
+    GlobalCustomerMainAgreement = get_global_customer_main_agreement_model(
+        AirTableBaseInfo.for_migrations(product_id)
+    )
+    global_customer_main_agreements = GlobalCustomerMainAgreement.all(
+        formula=AND(
+            EQUAL(FIELD("authorization_uk"), STR_VALUE(authorization_uk)),
+            OR(
+                EQUAL(FIELD("membership_id"), STR_VALUE(membership_or_customer_id)),
+                EQUAL(FIELD("customer_id"), STR_VALUE(membership_or_customer_id)),
+            ),
+        ),
+    )
+    return (
+        global_customer_main_agreements[0] if global_customer_main_agreements else None
+    )
+
+
+def get_global_customer_agreement_deployments_by_main_agreement(
+    product_id, main_agreement_id
+):
+    """
+    Retrieve the list of GlobalCustomerAgreementDeployment given the main agreement ID.
+
+    Args:
+        product_id (str): The ID of the product used to determine the AirTable base.
+        main_agreement_id (str): The ID of the main agreement.
+
+    Returns:
+        GlobalCustomerAgreementDeployment (list): The list of GlobalCustomerAgreementDeployment
+    """
+    GlobalCustomerAgreementDeployment = get_global_customer_agreement_deployment_model(
+        AirTableBaseInfo.for_migrations(product_id)
+    )
+    return GlobalCustomerAgreementDeployment.all(
+        formula=AND(
+            EQUAL(FIELD("main_agreement_id"), STR_VALUE(main_agreement_id)),
+        ),
+    )
+
+
+def get_global_customer_agreement_deployments_to_check(product_id):
```

Comment:
Probably let's intriduce a shortcut for `global customer agreement` --> gca? or just global_agreement?